aten/src/ATen/BatchingUtils.h

```diff
@@ -0,0 +1,39 @@
+#include <ATen/BatchedTensorImpl.h>
+
+namespace at {
+
+/*
+ * Utility functions used to implement batching rules.
+ *
+ * NB: All of these do NOT accept Tensors backed by BatchedTensor, unless
```

Comment:
Add some internal asserts to test for this invariant?

Reply:
Done